File: spring_java/src/main/java/dev/pollito/spring_java/sakila/film/domain/port/in/FindByIdPortInImpl.java
package dev.pollito.spring_java.sakila.film.domain.port.in;

import dev.pollito.spring_java.sakila.film.domain.model.Film;
import dev.pollito.spring_java.sakila.film.domain.port.out.FindByIdPortOut;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FindByIdPortInImpl implements FindByIdPortIn {
  private final FindByIdPortOut findByIdPortOut;

  @Override
  public Film findById(Integer id) {
    return findByIdPortOut.findById(id);
  }
}

File: spring_kotlin/src/main/kotlin/dev/pollito/spring_kotlin/sakila/film/domain/port/in/FindByIdPortInImpl.kt
package dev.pollito.spring_kotlin.sakila.film.domain.port.`in`

import dev.pollito.spring_kotlin.sakila.film.domain.model.Film
import dev.pollito.spring_kotlin.sakila.film.domain.port.out.FindByIdPortOut
import org.springframework.stereotype.Service

@Service
class FindByIdPortInImpl(private val findByIdPortOut: FindByIdPortOut) : FindByIdPortIn {
  override fun findById(id: Int): Film = findByIdPortOut.findById(id)
}
